### PR TITLE
Reuse resource mapper

### PR DESCRIPTION
**Pull request by YANYZP**
Please correct me if I have any misunderstanding of the requirements.

### CODE REVIEW

**Comment by YANYZP** [comment]
Relevant PRs:
https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/62
https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/pull/63
OC trace exporter:
https://github.com/census-ecosystem/opencensus-go-exporter-stackdriver/blob/master/trace_proto.go#L189

File: docs/resource-mapper.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# OpenTelemetry Resource mapper
+
+Turn the resource mapper in the metric exporter into a shared tool. It can be reused for both trace and metric exporters. 
+
+## Motivation
+
+OpenTelemetry Resource labels now can be inserted into trace spans. We want to allow Monitored
+Resources to be inserted as well. A shared resource mapper can enable trace exporter to convert
```
Comment (james-bebbington) [inline docs/resource-mapper.md:8]: "We want to allow Monitored Resources **(as derived from OpenTelemetry resources via `resourceToMonitoredResourcepb`)** to be inserted as well **(as well or instead?)**."

File: docs/resource-mapper.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+## Requirement
+
+For Trace exporter, the monitored resource labels need to be in the form of 
+<key = "g.co/r/<monitored_resource_type>/<monitored_resource_key>", value = "<monitored_resource_value>">
+However, we need to be aware that users can provide resources that cannot be converted to monitored resources.
+As for metric exporter, it does not need the prefix "g.co/r" when converting OT resources to monitored resources. 
+
+### Scheme
+
+We plan to move the resource mapper from metric exporter to tool package for reuse. As for the prefix "g.co/r", we would handle it in trace part. First, we would try to use the resource mapper to convert the resource. If it can be converted, then we would add the prefix "g.co/r" for all relevant key-value pairs. Otherwise, we keep the resource same and inject it into spans as before.
```
Comment (james-bebbington) [inline docs/resource-mapper.md:20]: Presumably this means we will only inject either Cloud Monitoring resource info (if can be mapped) or OT resource info (if can't be mapped).

I guess even if the OT resource info can be mapped there may be some additional resource attributes supplied by OT that you still want to send?

**Comment by james-bebbington** [comment]
I don't think we need to merge these docs. Have created https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/issues/88 to track this instead.

**Pull request closed without merging**